src/votekit/graphs.py

```diff
@@ -0,0 +1,162 @@
+                ]
+                missing_cands_perms = list(
+                    permutations(missing_cands, len(missing_cands))
+                )
+                frac_freq = ballot.weight / (len(missing_cands_perms))
+                for perm in missing_cands_perms:
+                    updated_rank = ballot.ranking + list(perm)
+                    updated_ballot = Ballot(
+                        ranking=updated_rank, weight=Fraction(frac_freq, 1)
+                    )
+                    updated_ballot_list.append(updated_ballot)
+            else:
+                updated_ballot_list.append(ballot)
+        return PreferenceProfile(ballots=updated_ballot_list)
+
+    # Helper functions to make pairwise comparison graph
+    def head2head_count(self, cand1, cand2) -> Fraction:
+        count = 0
+        ballots_list = self.profile.get_ballots()
+        for ballot in ballots_list:
+            rank_list = ballot.ranking
+            for s in rank_list:
+                if cand1 in s:
+                    count += ballot.weight
+                    break
+                elif cand2 in s:
+                    break
+        return Fraction(count)
+
+    def compute_pairwise_dict(self) -> dict:
+        pairwise_dict = {}  # {(cand_a, cand_b): freq cand_a is preferred over cand_b}
+        cand_pairs = list(combinations(self.candidates, 2))
+
+        for pair in cand_pairs:
+            cand_a, cand_b = pair[0], pair[1]
+            head_2_head_dict = {
+                (cand_a, cand_b): self.head2head_count(cand_a, cand_b),
+                (cand_b, cand_a): self.head2head_count(cand_b, cand_a),
+            }
+            max_pair = max(zip(head_2_head_dict.values(), head_2_head_dict.keys()))
+            pairwise_dict[max_pair[1]] = abs(
+                self.head2head_count(cand_a, cand_b)
+                - self.head2head_count(cand_b, cand_a)
+            )
+
+        return pairwise_dict
+
+    def generate_graph(self) -> nx.DiGraph:
+        G = nx.DiGraph()
+        G.add_nodes_from(self.candidates)
+        for e in self.pairwise_dict.keys():
+            G.add_edge(e[0], e[1], weight=self.pairwise_dict[e])
+        return G
+
+    def draw(self, show=True, outfile=None):
+        G = self.pairwise_graph
+
+        pos = nx.circular_layout(G)
+        nx.draw_networkx(
+            G,
+            pos,
+            with_labels=True,
+            node_size=500,
+            node_color="skyblue",
+            edgelist=list(),
+        )
+        nx.draw_networkx_edges(
+            G,
+            pos,
+            edgelist=G.edges,
+            width=1.5,
+            edge_color="b",
+            arrows=True,
+            alpha=1,
+            node_size=1000,
+            arrowsize=25,
+        )
+        edge_labels = {(i, j): G[i][j]["weight"] for i, j in G.edges()}
+        nx.draw_networkx_edge_labels(
+            G, pos, edge_labels=edge_labels, label_pos=0.5, font_size=10
+        )
+        # Out stuff

```

Comment:
tip: since you always want one of these to be true, it's worth considering combining them into one parameter

perhaps the default behavior is show=True, and iff outfile is passed that is used instead, otherwise show=False, outfile=None means this does nothing